table/compaction_merging_iterator.cc

```diff
@@ -47,6 +63,14 @@ class CompactionMergingIterator : public InternalIterator {
   }
 
   ~CompactionMergingIterator() override {
+    if (internal_stats_) {
+      assert(num_sorted_runs_recorded_ == range_tombstone_iters_.size());
```

Comment:
what's this for?

Reply:
The `assert(num_sorted_runs_recorded_ == range_tombstone_iters_.size());` is related to this comment in the constructor that I added

```
      // The size of children_ or range_tombstone_iters_ (n) should not change
      // but to be safe, we can record the size here so we decrement by the
      // correct amount at destruction time
      num_sorted_runs_recorded_ = n;
```

Reply:
For `children_`, I don't think we need to worry about those for counting the number of sorted runs. Even though technically they are `InternalIterator`s, we would not expect to have a `CompactionMergingIterator` within a `CompactionMergingIterator` for instance

Reply:
> I don't think we need to worry about those for counting the number of sorted runs.

Sorry I'm still a bit confused about this: I thought you would want `assert(num_sorted_runs_recorded_ == children_.size())` too just to match what you said in the comment "The size of children_ or..."

Reply:
We could do `assert(num_sorted_runs_recorded_ == children_.size())` as well. That would serve the same purpose for me.